engine-nextgen/src/test/kotlin/graphql/nadel/validation/NadelFieldValidationTest.kt

```diff
@@ -62,6 +64,86 @@ class NadelFieldValidationTest : DescribeSpec({
             assert(error.argument.name == "world")
         }
 
+        it("passes if overall argument value is more strict") {
+            val fixture = NadelValidationTestFixture(
+                    overallSchema = mapOf(
+                            "test" to """
+                        type Query {
+                            echo(world: Boolean!): String
```

Comment:
Add a test for `world: [Boolean]` in overall and `World: [Boolean!]` in underlying

It should fail but I don't think it will with the current impl